### PR TITLE
chore: Improve error message when URLs don't start with `prisma://` when using dataproxy

### PR DESCRIPTION
**Pull request by jacobhq**
This helps people with the headache I experienced in #11526.

### CODE REVIEW

File: packages/engine-core/src/data-proxy/DataProxyEngine.ts [hunk 1]
```diff
@@ -237,7 +237,7 @@ export class DataProxyEngine extends Engine {
     const { protocol, host, searchParams } = url
 
     if (protocol !== 'prisma:') {
-      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol', {
+      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol. If you are not using the Data Proxy, remove it from your schema', {
         clientVersion: this.clientVersion,
       })
```
Comment (Jolg42) [inline packages/engine-core/src/data-proxy/DataProxyEngine.ts:242]:
```suggestion
      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol. If you are not using the Data Proxy, remove the `dataProxy` from the `previewFeatures` from your schema', {
        clientVersion: this.clientVersion,
      })
```

**Comment by millsp** [comment]
Hey @jacobhq, I'm curious to know how you ran into this and why it has caused confusion. The `dataproxy` has to be enabled in two ways, the one is via the preview features, and the second is via `PRISMA_CLIENT_ENGINE_TYPE=dataproxy`. Have you forgotten removing the `PRISMA_CLIENT_ENGINE_TYPE=dataproxy` env var? If that is the case, that might be a more appropriate piece of info to display in that message and given that `dataProxy` is going to move out of preview in the future.

**Comment by jacobhq** [comment]
@millsp I had copied the schema from another project where I was using the dataproxy, into a project where I wasn't, and hadn't realised. I wrote loads more in #11526 [here](https://github.com/prisma/prisma/issues/11526#issuecomment-1053698253)

**Comment by millsp** [comment]
Thanks. Could you please share that schema with me?

**Comment by jacobhq** [comment]
I did not have that env var set. Here is the schema:
```
datasource db {
  provider             = "postgresql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("DATABASE_MIGRATE_URL")
  referentialIntegrity = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["dataproxy", "referentialIntegrity"]
}

model User {
  id           Int            @id @default(autoincrement())
  WishlistItem WishlistItem[]
}

model Product {
  id           Int            @id @default(autoincrement())
  name         String
  price        Int
  salePrice    Int            @default(0)
  imageUrl     String
  flag         String?
  description  String?
  buyUrl       String
  marketplace  String?
  WishlistItem WishlistItem[]
}

model WishlistItem {
  id        Int     @id @default(autoincrement())
  userId    Int
  productId Int
  Product   Product @relation(fields: [productId], references: [id])
  User      User    @relation(fields: [userId], references: [id])
}
```

File: packages/engine-core/src/data-proxy/DataProxyEngine.ts [hunk 2]
```diff
@@ -237,7 +237,7 @@ export class DataProxyEngine extends Engine {
     const { protocol, host, searchParams } = url
 
     if (protocol !== 'prisma:') {
-      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol', {
+      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol. If you are not using the Data Proxy, remove the `dataProxy` from the `previewFeatures` from your schema', {
```
Comment (aqrln) [inline packages/engine-core/src/data-proxy/DataProxyEngine.ts:240]:
```suggestion
      throw new InvalidDatasourceError('Datasource URL should use prisma:// protocol. If you are not using the Data Proxy, remove the `dataProxy` from the `previewFeatures` in your schema and ensure that `PRISMA_CLIENT_ENGINE_TYPE` environment variable is not set to `dataproxy`.', {
```

Comment (janpio) [reply packages/engine-core/src/data-proxy/DataProxyEngine.ts:240]: We will need to remember to update this later when/if we update to another mechanism to activate the data proxy client.

**Review by millsp — APPROVED** [review]

**Pull request merged**